streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -124,10 +124,38 @@ object ZTransducer {
     }
 
   /**
-   * Creates a transducer accumulating incoming values into lists of maximum size `n`.
+   * Creates a transducer accumulating incoming values into chunks of maximum size `n`.
    */
-  def collectAllN[I](n: Long): ZTransducer[Any, Nothing, I, List[I]] =
-    foldUntil[I, List[I]](Nil, n)((list, element) => element :: list).map(_.reverse).filter(_.nonEmpty)
+  def collectAllN[I](n: Long): ZTransducer[Any, Nothing, I, Chunk[I]] =
+    ZTransducer {
+
+      def go(in: Chunk[I], builder: ChunkBuilder[I], size: Long): (ChunkBuilder[Chunk[I]], ChunkBuilder[I], Long) =
```

Comment:
You can append `in` to `builder` all at once instead of appending elements one by one. It is faster and would save some lines of code as well

Reply:
Good idea, I was wrapping my mind around transducer internals while writing this :sweat_smile:
In that approach we need to handle case when appending the `in` would produce Chunk that has more elements than `N` by splitting it accordingly and carrying over leftover. I will check `Chunk#splitAt` but afaik it should be better than current impl.